Change in spack failure of automated build

@hainest this is for you! Slack is on my other computer and I'm too lazy to walk 20 feet at the moment :)

```bash
2022-02-02T03:34:23.4886639Z ==> Fetching https://mirror.spack.io/_source-cache/archive/7f/7f6fb9149b1673d38d9178a0d3e0fb8a1ec4f53a9f4c2ff89469609879641177.tar.bz2
2022-02-02T03:34:23.4887050Z ==> Ran patch() for elfutils
2022-02-02T03:34:23.4980124Z ==> elfutils: Executing phase: 'autoreconf'
2022-02-02T03:34:23.5135765Z ==> elfutils: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 77:
    '/tmp/root/spack-stage/spack-stage-elfutils-0.186-64o7eqipg4bxe52irdh5qohwsuk63o5b/spack-src/configure' '--prefix=/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.0.1/elfutils-0.186-64o7eqipg4bxe52irdh5qohwsuk63o5b' '--without-bzlib' '--without-lzma' '--with-zlib=/opt/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.0.1/zlib-1.2.11-oski6wpiq4xvenaqdds4anr4cnutrswy' 'LDFLAGS=-Wl,--no-as-needed -L/usr/lib -lintl' '--disable-debuginfod' '--disable-libdebuginfod'

2 errors found in build log:
     13    checking whether make supports nested variables... (cached) yes
     14    checking build system type... x86_64-pc-linux-gnu
     15    checking host system type... x86_64-pc-linux-gnu
     16    checking whether make supports the include directive... yes (GNU sty
           le)
     17    checking for gcc... /opt/spack/lib/spack/env/gcc/gcc
     18    checking whether the C compiler works... no
  >> 19    configure: error: in `/tmp/root/spack-stage/spack-stage-elfutils-0.1
           86-64o7eqipg4bxe52irdh5qohwsuk63o5b/spack-src':
  >> 20    configure: error: C compiler cannot create executables
     21    See `config.log' for more details

See build log for details:
  /tmp/root/spack-stage/spack-stage-elfutils-0.186-64o7eqipg4bxe52irdh5qohwsuk63o5b/spack-build-out.txt

==> Updating view at /opt/dyninst-env/.spack-env/view
```

It looks like something has changed in spack so our base container failed today. Unlike last time I don't see any changes to the failing package - indeed it's the same release from last year (and the build from yesterday worked):

```bash
==> Installing elfutils-0.186-iosmb7qegoihv4h2wsrftodkoykcjyd4
==> No binary for elfutils-0.186-iosmb7qegoihv4h2wsrftodkoykcjyd4 found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/7f/7f6fb9149b1673d38d9178a0d3e0fb8a1ec4f53a9f4c2ff89469609879641177.tar.bz2
==> Ran patch() for elfutils
==> elfutils: Executing phase: 'autoreconf'
==> elfutils: Executing phase: 'configure'
==> elfutils: Executing phase: 'build'
==> elfutils: Executing phase: 'install'
==> elfutils: Successfully installed elfutils-0.186-iosmb7qegoihv4h2wsrftodkoykcjyd4
  Fetch: 0.66s.  Build: 1m 4.47s.  Total: 1m 5.13s.
```
It seems to be the same source code from the cache, so it can't be that (7f6fb9149b1673d38d9178a0d3e0fb8a1ec4f53a9f4c2ff89469609879641177). The only change in elfutils was adding a patch for Mac, which only should be applied for [gcc 11.2.0](https://github.com/spack/spack/commit/80a3f01bbbb6304ee4791720279b29d59c799ec7#diff-5089344542ef0a05b823ba5a050403c20febec624522362959d5be1de9a07c59). 

One thing I don't quite understand is why it's using `/opt/spack/lib/spack/env/gcc/gcc` when we have the gcc on the system as a virtual dependency. It could be that if it's trying to use the default, 11.2.0, the patch would be applied and it might result in this "does not create executables" business. I'm going to ping @alalazo to see if he has insight to why:

1. the install might be using some different version of gcc installed with spack (which would default to the last 11.2.0 if relecant)
2. and if so, why this patch https://github.com/spack/spack/commit/80a3f01bbbb6304ee4791720279b29d59c799ec7#diff-a108a05bff2c1ef067255bebfad8328ab4d999225ed88631766160c5f77b7a7c is being applied without checking that we are on a mac possibly resulting in this issue?

And just for kicks and giggles I'm going to re-run in case spack was just being a :hankey: face. If it errors again, @alalazo let me know what you'd like to see in the build to debug and I'll make it so!

--- a/.github/workflows/base-containers.yaml
+++ b/.github/workflows/base-containers.yaml
@@ -9,8 +9,6 @@ on:
   # Publish packages on release
   release:
     types: [published] 
-    
-  pull_request: []
 
   # On push to main we build and deploy images
   push: 

--- a/.github/workflows/base-containers.yaml
+++ b/.github/workflows/base-containers.yaml
@@ -10,6 +10,8 @@ on:
   release:
     types: [published] 
     
+  pull_request: []
+
   # On push to main we build and deploy images
   push: 
     branches:

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -66,12 +66,8 @@ RUN apt-get install -y software-properties-common && \
     apt-get update && \
     apt-get install -y gcc-11 g++-11 && \
     update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 70 --slave /usr/bin/g++ g++ /usr/bin/g++-9 --slave /usr/bin/gcov gcov /usr/bin/gcov-9 --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-9 --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-9 && \
-    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 110 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-11 --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-11 --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-11; \
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 110 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-11 --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-11 --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-11;
    
-    # This is installed just for Ubuntu 20.04 and failing spack builds.
-    # Ideally we can select to not use it with external find
-    apt remove -y gettext
-
 # Install spack
 WORKDIR /opt
 RUN git clone --depth 1 https://github.com/spack/spack
@@ -85,7 +81,7 @@ RUN python3 -m pip install botocore boto3 && \
     spack gpg trust key.pub
 
 # Find packages already installed on system, e.g. autoconf
-RUN spack external find && \
+RUN spack external find gcc@11.0.1 autoconf bzip2 git tar xz && \
     spack config add 'packages:all:target:[x86_64]' && \
     # Install a new CMake (Tim originally wanted 3.17.1 but spack doesn't have it)
     spack install cmake@${CMAKE_VERSION} perl

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -66,7 +66,11 @@ RUN apt-get install -y software-properties-common && \
     apt-get update && \
     apt-get install -y gcc-11 g++-11 && \
     update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 70 --slave /usr/bin/g++ g++ /usr/bin/g++-9 --slave /usr/bin/gcov gcov /usr/bin/gcov-9 --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-9 --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-9 && \
-    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 110 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-11 --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-11 --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-11;
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 110 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-11 --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-11 --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-11; \
+   
+    # This is installed just for Ubuntu 20.04 and failing spack builds.
+    # Ideally we can select to not use it with external find
+    apt remove -y gettext
 
 # Install spack
 WORKDIR /opt

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -81,7 +81,7 @@ RUN python3 -m pip install botocore boto3 && \
     spack gpg trust key.pub
 
 # Find packages already installed on system, e.g. autoconf
-RUN spack external find gcc@11.0.1 autoconf bzip2 git tar xz && \
+RUN spack external find gcc@11.0.1 autoconf bzip2 git tar xz perl && \
     spack config add 'packages:all:target:[x86_64]' && \
     # Install a new CMake (Tim originally wanted 3.17.1 but spack doesn't have it)
     spack install cmake@${CMAKE_VERSION} perl

